mesh/mesh.hpp

```diff
@@ -1010,6 +1014,44 @@ class Mesh
    /// store these in the Mesh::attributes and Mesh::bdr_attributes arrays.
    virtual void SetAttributes();
 
+   /// @brief Prepares a marker array corresponding to an array of element
+   /// attributes
+   ///
+   /// @param max_attr Number of entries to create in the @a marker array
+   /// @param attrs    An array of attribute numbers which should be activated
+   /// @param marker   Output array indicating active/inactive attributes
+   ///
+   /// The marker array will be of size max_attr and it will contain only zeroes
+   /// and ones. Ones indicate which attribute numbers are present in the attrs
+   /// array. In the special case when attrs has a single entry equal to -1 the
+   /// marker array will contain all ones.
+   void AttrToMarker(int max_attr, const Array<int> &attrs,
+                     Array<int> &marker) const;
```

Comment:
Does this need to be a (non-static) `Mesh` member function? To me it seems like it could either be a free function or a static member function

Reply:
Right, this should be static.

Reply:
Note that `FiniteElementSpace` also has static member functions `MarkerToList` and `ListToMarker`, which do essentially the same thing.

Maybe these should be factored out to a general-purpose utility function, since the functionality is not specific to `Mesh` (nor to `FiniteElementSpace`).

Reply:
`FiniteElementSpace::MarkerToList` and `FiniteElementSpace::ListToMarker` operate on the host. Are you worried that in the future this might change?

Reply:
I'm not worried about anything I just see these as very different cases and I like having separate implementations in these two classes because DoFs are naturally associated with the `FiniteElementSpace` and domain/boundary regions with the `Mesh`.

Reply:
I see what you mean. Yeah, seems not worth it to unify them (especially since the attribute indices have to be shifted by -1. But besides for this difference, the implementations are actually the same).